pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/timeseries/TimeSeriesExchangeReceiveOperator.java

```diff
@@ -218,6 +244,10 @@ private TimeSeriesBlock getNextBlockNoAggregation()
       mergeStats(aggregatedStats, blockToMerge.getMetadata());
     }
     Preconditions.checkNotNull(timeBuckets, "Time buckets is null in exchange receive operator");
-    return new TimeSeriesBlock(timeBuckets, timeSeriesMap, aggregatedStats);
+    TimeSeriesBlock resultBlock = new TimeSeriesBlock(timeBuckets, timeSeriesMap, aggregatedStats);
+    if (timeoutException != null) {
+      resultBlock.addToExceptions(timeoutException);
```

Comment:
@shauryachats : in a future PR let's also update exception handling in the open source timeseries plugin.

cc: @raghavyadav01 : after this PR, if you are combining time series blocks in operations such as divideSeries, you should also copy over exceptions from any of the blocks and pass them to the resulting block.